lib/EbookTag.php

```diff
@@ -17,4 +18,45 @@ protected function GetUrl(): string{
 
 		return $this->_Url;
 	}
+
+	// *******
+	// METHODS
+	// *******
+	public function Validate(): void{
+		$error = new Exceptions\ValidationException();
+
+		if(strlen($this->Name) > EBOOKS_MAX_STRING_LENGTH){
+			$error->Add(new Exceptions\StringTooLongException('Ebook tag: '. $this->Name));
+		}
+
+		if($error->HasExceptions){
+			throw $error;
+		}
+	}
+
+	public function Create(): void{
+		$this->Validate();
+
+		Db::Query('
+			INSERT into Tags (Name)
+			values (?)
+		', [$this->Name]);
+		$this->TagId = Db::GetLastInsertedId();
+	}
+
+	public static function GetOrCreate(EbookTag $ebookTag): EbookTag{
```

Comment:
Check it out now, and see if it's more appropriate. Because of my variable names, it doesn't read as cleanly as your example.

I agree the pattern before wasn't great. I was trying to get away with a static factory without creating a whole separate class.